frontend/src/container/JoinWorkspace.jsx

```diff
@@ -0,0 +1,181 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { translate } from 'react-i18next'
+import { Link } from 'react-router-dom'
+import escapeRegExp from 'lodash/escapeRegExp'
+
+import {
+  PageWrapper,
+  PageTitle,
+  PageContent,
+  CUSTOM_EVENT,
+  BREADCRUMBS_TYPE,
+  TracimComponent,
+  SPACE_TYPE,
+  SPACE_TYPE_LIST,
+  SUBSCRIPTION_TYPE,
+  IconButton,
+  TextInput
+} from 'tracim_frontend_lib'
+
+import {
+  newFlashMessage,
+  setBreadcrumbs,
+  setHeadTitle,
+  setWorkspaceSubscriptionList
+} from '../action-creator.sync.js'
+import {
+  PAGE
+} from '../util/helper.js'
+import { getWorkspaceSubscriptions, joinWorkspace, subscribeToWorkspace } from '../action-creator.async.js'
+
+export class JoinWorkspace extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.state = { filter: () => true }
+
+    props.registerCustomEventHandlerList([
+      { name: CUSTOM_EVENT.ALL_APP_CHANGE_LANGUAGE, handler: this.handleAllAppChangeLanguage }
+    ])
+  }
+
+  async componentDidMount () {
+    this.handleAllAppChangeLanguage()
+    await this.loadWorkspaceSubscriptions()
+  }
+
+  async loadWorkspaceSubscriptions () {
+    const { props } = this
+
+    const fetchSubscriptionList = await props.dispatch(getWorkspaceSubscriptions(props.user.userId))
+
+    if (fetchSubscriptionList.status !== 200) {
+      props.dispatch(newFlashMessage(props.t('An error has happened'), 'warning'))
+      return
+    }
+    props.dispatch(setWorkspaceSubscriptionList(fetchSubscriptionList.json))
+  }
+
+  handleAllAppChangeLanguage () {
+    const { props } = this
+    this.buildBreadcrumbs()
+    props.dispatch(setHeadTitle(props.t('Join a space')))
+  }
+
+  buildBreadcrumbs () {
+    const { props } = this
+
+    props.dispatch(setBreadcrumbs([{
+      link: <Link to={PAGE.HOME}><i className='fa fa-home' />{props.t('Home')}</Link>,
+      type: BREADCRUMBS_TYPE.CORE
+    }, {
+      link: <Link to={PAGE.JOIN_WORKSPACE}>{props.t('Join a space')}</Link>,
+      type: BREADCRUMBS_TYPE.CORE
+    }]))
+  }
+
+  async joinWorkspace (workspaceId) {
+    const { props } = this
+    await props.dispatch(joinWorkspace(workspaceId, props.user.userId))
+    props.history.push(PAGE.WORKSPACE.DASHBOARD(workspaceId))
+  }
+
+  createRequestComponent (workspace) {
+    const { props } = this
+
+    const subscription = props.workspaceSubscriptionList.find(s => s.workspace.workspace_id === workspace.id)
+
+    if (subscription !== undefined) {
+      var text = 'Unknown request state'
+      var icon = 'question'
```

Comment:
don't use var keyword, prefer const. Or, `let` here since you reassign it
also, `text` isn't translated

Reply:
I didn't translate the text as this is more a "bug detection" to avoid breaking the interface rather than a string that should usually be displayed to users.